resources/assets/js/videos/videoContainer.vue

```diff
@@ -560,7 +580,19 @@ export default {
                 .then(this.maybeFocusInitialAnnotation)
                 .then(this.maybeInitCurrentTime);
 
-            this.video.src = this.videoFileUri.replace(':id', video.id);
+            let self = this;
+            fetch(this.videoFileUri.replace(':id', video.id))
+                .then((res) => {
+                    res.blob().then(function (blob) {
+                        let urlCreator = window.URL || window.webkitURL;
+                        self.video.src = urlCreator.createObjectURL(blob);
+                    });
+                })
+                .catch((e) => {
+                    // Access on non-CORS enabled file causes TypeError
+                    this.hasCrossOriginError = e instanceof TypeError;
+                    self.video.src = self.videoFileUri.replace(':id', video.id);
+                });
```

Comment:
This tries to download the whole video. Usually the video is fetched dynamically through a series of HTTP range requests, which is crucial here.

You need another method to detect CORS. Maybe create a small canvas element and try to read some data from the video into it?